Mathlib/Data/Matrix/Basic.lean

```diff
@@ -410,7 +442,8 @@ theorem diagonal_zero [Zero α] : (diagonal fun _ => 0 : Matrix n n α) = 0 := b
 
 @[simp]
 theorem diagonal_transpose [Zero α] (v : n → α) : (diagonal v)ᵀ = diagonal v := by
-  ext (i j)
+  ext
+  rename_i i j -- Porting note: `ext` does not like `Matrix.ext` introducing two vars at once.
```

Comment:
I thought you had a solution to this by adding a weird `@[ext]` lemma for matrices that amounted to "two matrices are equal if each row is equal", which meant that lean would apply `funext` after `matrix.ext` and do the right thing?

Reply:
Oh I miss understood you. I'll change it to that option!

Reply:
Thanks! I preferred it because it looks easier to undo later if we make a fix to core / std.